src/main.py

```diff
@@ -20,56 +51,69 @@
     Application.TRANSIT: environ["TRANSIT_SLACK_USER_IDS"],
-    Application.UPLIFT: environ["UPLIFT_SLACK_USER_IDS"],
+    Application.VOLUME: environ["VOLUME_SLACK_USER_IDS"],
 }
 application_slack_hook_mapping = {
     Application.COURSEGRAB: environ["SLACK_HOOK_COURSEGRAB_URL"],
     Application.EATERY: environ["SLACK_HOOK_EATERY_URL"],
     Application.TRANSIT: environ["SLACK_HOOK_TRANSIT_URL"],
-    Application.UPLIFT: environ["SLACK_HOOK_UPLIFT_URL"],
+    Application.VOLUME: environ["SLACK_HOOK_VOLUME_URL"],
 }
 
-for test_group in test_groups:
-    slack_message_text = "\tRunning tests for {}...\n".format(test_group.name)
-    test_group_failures = 0
-    num_test_group_tests = len(test_group.tests)
-
-    diagnostic_text = ""
-    for test in test_group.tests:
-        test_status = SUCCESS_STATUS  # default to printing success
-        test_result = test.get_result()
-        if test_result != Result.SUCCESS:
-            test_group_failures += 1
-            # Mark that there is an error in this pod
-            test_group.slack_message.should_send = True
-        diagnostic_text += "[{}] - {}\n".format(test.name, test_result.name)
-    slack_message_text += "> ```\n{}```\n".format(diagnostic_text)
-
-    if test_group_failures:
-        passed_tests = num_test_group_tests - test_group_failures
-        slack_message_text += "\t*Summary: `{}/{}` tests passed!* ".format(passed_tests, num_test_group_tests)
-        # Tag appropriate users
-        user_ids = environ["ADMIN_SLACK_USER_IDS"]
-        if test_group.application in application_user_id_mapping:
-            user_ids += ", " + application_user_id_mapping[test_group.application]
-        slack_message_text += "cc {}!".format(user_ids)
-    else:
-        slack_message_text = "*`{0}/{0}` tests passed :white_check_mark:*".format(num_test_group_tests)
-    test_group.slack_message.text = slack_message_text
-
-    # Always print output for debugging purposes
-    print(test_group.slack_message.text)
+
+for i in range(len(test_groups)):
+    if test_config[i]!="0":  # Only test enabled or previously failing test groups
+        test_group=test_groups[i]
+        slack_message_text = "\tRunning tests for {}...\n".format(test_group.name)
+        test_group_failures = 0
+        num_test_group_tests = len(test_group.tests)
+
+        diagnostic_text = ""
+        for test in test_group.tests:
+            test_status = SUCCESS_STATUS  # default to printing success
+            test_result = test.get_result()
+            if test_result != Result.SUCCESS:
+                test_group_failures += 1
+                # Mark that there is an error in this pod
+                test_group.slack_message.should_send = True
+            diagnostic_text += "[{}] - {}\n".format(test.name, test_result.name)
+        slack_message_text += "> ```\n{}```\n".format(diagnostic_text)
+
+        if test_group_failures:
+            test_config[i]="2" # The test group has failed
```

Comment:
this is good! but it would be useful to suppress back to back successes as well in the future and also inform us of any new failures / fixed failures (if I have 2 failures and the next test check has only 1 failure meaning one was fixed or 3 failures meaning an additional failure occurred).

Reply:
successes don't ping people so i think it might be good to have to show the bot is still running